openapi/data/validate.py

```diff
@@ -13,30 +13,39 @@ class ValidatedData:
 def validate(schema, data, strict=True):
     """Validate a dictionary of data with a given dataclass
     """
+    data = data.copy()
     errors = {}
     cleaned = {}
     for field in schema.__dataclass_fields__.values():
         try:
             required = field.metadata.get(REQUIRED)
-            if field.name not in data and DEFAULT not in field.metadata:
-                if required and strict:
-                    raise ValidationError(field.name, 'required')
-                continue
-
             validator = field.metadata.get(VALIDATOR)
-            default = field.metadata.get(DEFAULT)
-            value = data.get(field.name, default)
+            if DEFAULT in field.metadata:
+                data.setdefault(field.name, field.metadata[DEFAULT])
+
+            if field.name not in data and required and strict:
+                raise ValidationError(field.name, 'required')
+
+            ops = field.metadata.get(OPS, ())
+            for name in [field.name] + [f'{field.name}:{op}' for op in ops]:
```

Comment:
We need to use this iteration at least in two places. Lets create a function for it.
I propose something along these lines:
```python
def field_ops(field):
    yield field.name
    for op in field.metadata.get(OPS, ()):
        yield f'{field.name}:{op}'
```